lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+        return failure();
+  }
+
+  jemit.endList(); // edges
+
+  jemit.endObject();
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// TaskletNode
+//===----------------------------------------------------------------------===//
+
+// Temporary auto-lifting. Will be included into DaCe
+LogicalResult liftToPython(TaskletNode &op, JsonEmitter &jemit) {
+  int numOps = 0;
+  Operation *firstOp = nullptr;
+
+  for (Operation &oper : op.body().getOps()) {
+    if (numOps >= 2)
+      return failure();
+    if (numOps == 0)
+      firstOp = &oper;
+    ++numOps;
+  }
+
+  AsmState state(op);
+
+  if (dyn_cast<arith::AddFOp>(firstOp) || dyn_cast<arith::AddIOp>(firstOp)) {
+    std::string nameArg0;
+    llvm::raw_string_ostream nameArg0Stream(nameArg0);
+    op.getArgument(0).printAsOperand(nameArg0Stream, state);
+    nameArg0.erase(0, 1); // Remove %-sign
+
+    std::string nameArg1;
+    llvm::raw_string_ostream nameArg1Stream(nameArg1);
+    op.getArgument(1).printAsOperand(nameArg1Stream, state);
+    nameArg1.erase(0, 1); // Remove %-sign
+
+    jemit.printKVPair("string_data", "__out = " + nameArg0 + " + " + nameArg1);
+    jemit.printKVPair("language", "Python");
+    return success();
+  }
+
+  if (dyn_cast<arith::MulFOp>(firstOp) || dyn_cast<arith::MulIOp>(firstOp)) {
+    std::string nameArg0;
+    llvm::raw_string_ostream nameArg0Stream(nameArg0);
+    op.getArgument(0).printAsOperand(nameArg0Stream, state);
+    nameArg0.erase(0, 1); // Remove %-sign
+
+    std::string nameArg1;
+    llvm::raw_string_ostream nameArg1Stream(nameArg1);
+    op.getArgument(1).printAsOperand(nameArg1Stream, state);
+    nameArg1.erase(0, 1); // Remove %-sign
+
+    jemit.printKVPair("string_data", "__out = " + nameArg0 + " * " + nameArg1);
+    jemit.printKVPair("language", "Python");
+    return success();
+  }
+
+  if (arith::ConstantFloatOp oper = dyn_cast<arith::ConstantFloatOp>(firstOp)) {
+    std::string val = std::to_string(oper.value().convertToDouble());
+    jemit.printKVPair("string_data", "__out = dace.float64(" + val + ")");
+    jemit.printKVPair("language", "Python");
+    return success();
+  }
+
+  if (arith::ConstantIntOp oper = dyn_cast<arith::ConstantIntOp>(firstOp)) {
+    std::string val = std::to_string(oper.value());
+    jemit.printKVPair("string_data", "__out = dace.int64(" + val + ")");
+    jemit.printKVPair("language", "Python");
+    return success();
+  }
+
+  return failure();
+}
+
+LogicalResult translateTaskletToSDFG(TaskletNode &op, JsonEmitter &jemit) {
+  jemit.startObject();
+  jemit.printKVPair("type", "Tasklet");
+  jemit.printKVPair("label", op.sym_name());
+
+  jemit.startNamedObject("attributes");
+  jemit.printKVPair("label", op.sym_name());
+
+  if (!containsAttr(*op, "instrument"))
+    jemit.printKVPair("instrument", "No_Instrumentation");
+
+  jemit.startNamedObject("code");
+
+  AsmState state(op);
+
+  if (liftToPython(op, jemit).failed()) {
```

Comment:
Document this branch, it's not a normal "if", the computation inside is substantial.